core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -613,7 +613,12 @@ class DAGScheduler(
       properties: Properties): Unit = {
     val start = System.nanoTime
     val waiter = submitJob(rdd, func, partitions, callSite, resultHandler, properties)
-    Await.ready(waiter.completionFuture, atMost = Duration.Inf)
+    // Note: Do not call Await.ready(future) because that calls `scala.concurrent.blocking`,
```

Comment:
BlockManager#doPut() does this on line 891:
```
Await.ready(replicationFuture, Duration.Inf)
```
Should that be replaced as well ?